tests/CMakeLists.txt

```diff
@@ -26,21 +26,42 @@ if(VEXCL_TEST_COVERAGE)
         )
 endif()
 
-function(add_vexcl_test TEST_NAME TEST_SOURCE)
+# Add one test with backend
+function(_add_vexcl_test TEST_NAME TEST_SOURCE BACKEND)
```

Comment:
This fails for `multiple_objects` target, which has more than one source file. Second source file gets picked for `BACKEND`.

Reply:
That was a good idea; I also then could use `${ARGN}` to handle multiple sources correctly. :)

Reply:
My first thought was to add keyword arguments, but this is better.